app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class LikesController < ApplicationController
+  def create
+    post = Post.find(params[:post_id])
+    Like.create(user: current_user, post: post)
+    redirect_back(fallback_location: user_post_path(post.author, post))
+  end
```

Comment:
- [ ] When creating your comment, you are not checking for potential errors using the `if else` statements as you did in your post_controller, i kindly suggest you do that so that incase there is an error while creating a comment, an error message can be displayed and handled properly, see the code snippet below as a guide
- code snippet
```ruby
@like =  Like.create(user: current_user, post: post)
 if @like.save
      flash[:notice] = 'Like created successfully'
       redirect_back(fallback_location: user_post_path(post.author, post))
      else
      flash.now[:error] = "Error: Couldn't create liket"
      render :new, status: :unprocessable_entity
    end
```